src/ethereum/tangerine_whistle/vm/instructions/environment.py

```diff
@@ -0,0 +1,434 @@
+"""
+Ethereum Virtual Machine (EVM) Environmental Instructions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. contents:: Table of Contents
+    :backlinks: none
+    :local:
+
+Introduction
+------------
+
+Implementations of the EVM environment related instructions.
+"""
+
+from ethereum.base_types import U256, Uint
+from ethereum.utils.numeric import ceil32
+from ethereum.utils.safe_arithmetic import u256_safe_add, u256_safe_multiply
+
+from ...state import get_account
+from ...utils.address import to_address
+from ...vm.error import OutOfGasError
+from ...vm.memory import extend_memory, memory_write
+from .. import Evm
+from ..gas import (
+    GAS_BALANCE,
+    GAS_BASE,
+    GAS_COPY,
+    GAS_EXTERNAL,
+    GAS_VERY_LOW,
+    calculate_gas_extend_memory,
+    subtract_gas,
+)
+from ..stack import pop, push
+
+
+def address(evm: Evm) -> None:
+    """
+    Pushes the address of the current executing account to the stack.
+
+    Parameters
+    ----------
+    evm :
+        The current EVM frame.
+
+    Raises
+    ------
+    ethereum.tangerine_whistle.vm.error.OutOfGasError
+        If `evm.gas_left` is less than `2`.
+    """
+    evm.gas_left = subtract_gas(evm.gas_left, GAS_BASE)
+    push(evm.stack, U256.from_be_bytes(evm.message.current_target))
+
+    evm.pc += 1
+
+
+def balance(evm: Evm) -> None:
+    """
+    Pushes the balance of the given account onto the stack.
+
+    Parameters
+    ----------
+    evm :
+        The current EVM frame.
+
+    Raises
+    ------
+    ethereum.tangerine_whistle.vm.error.StackUnderflowError
+        If `len(stack)` is less than `1`.
+    ethereum.tangerine_whistle.vm.error.OutOfGasError
+        If `evm.gas_left` is less than `20`.
+    """
+    # TODO: There are no test cases against this function. Need to write
+    # custom test cases.
+    evm.gas_left = subtract_gas(evm.gas_left, GAS_BALANCE)
```

Comment:
How would you feel about backporting the new constants to `frontier` and `homestead`? I think that would improve the diff readability.

Reply:
Sounds good to me! Will do this in a separate commit.